app/src/main/java/ch/sdp/vibester/fragment/GameSetupFragment.kt

```diff
@@ -0,0 +1,206 @@
+            if (view.findViewById<ConstraintLayout>(R.id.chooseGenre).visibility == View.VISIBLE) {
+                toggleViewsVisibility(goneView = view.findViewById<ConstraintLayout>(R.id.chooseGenre),
+                    visibleView = view.findViewById<LinearLayout>(R.id.chooseGame))
+            } else if (view.findViewById<RelativeLayout>(R.id.chooseSetting).visibility == View.VISIBLE) {
+                toggleViewsVisibility(goneView =view.findViewById<RelativeLayout>(R.id.chooseSetting),
+                    visibleView = view.findViewById<ConstraintLayout>(R.id.chooseGenre))
+            }
+        }
+    }
+
+    private fun toggleViewsVisibility(goneView: View, visibleView:View){
+        goneView.visibility = View.GONE
+        visibleView.visibility = View.VISIBLE
+    }
+
+    private fun setSpinnerListener(view: View, ctx: Context, spinnerId: Int, resourceId: Int) {
+        val spinner: Spinner = view.findViewById(spinnerId)
+        val adapter = ArrayAdapter.createFromResource(
+            ctx, resourceId, android.R.layout.simple_spinner_item
+        )
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        spinner.adapter = adapter
+        spinner.onItemSelectedListener = this
+    }
+
+    override fun onItemSelected(parent: AdapterView<*>, view: View?, position: Int, id: Long) {
+        if (parent.id == R.id.difficulty_spinner) {
+            difficulty = parent.getItemAtPosition(position).toString()
+        } else if (parent.id == R.id.size_spinner) {
+            gameSize = parent.getItemAtPosition(position).toString()
+            when (gameSize) {
+                "One" -> gameManager.setGameSize(1)
+                "Two" -> gameManager.setGameSize(2)
+                "Three" -> gameManager.setGameSize(3)
+                "Four" -> gameManager.setGameSize(4)
+                "Five" -> gameManager.setGameSize(5)
+                "Six" -> gameManager.setGameSize(6)
+                "Seven" -> gameManager.setGameSize(7)
+                "Eight" -> gameManager.setGameSize(8)
+                "Nine" -> gameManager.setGameSize(9)
+                "Ten" -> gameManager.setGameSize(10)
+            }
+        }
+    }
+
+    override fun onNothingSelected(parent: AdapterView<*>) {
+        if (parent.id == R.id.difficulty_spinner) {
+            difficulty = R.string.easy.toString()
+        } else if (parent.id == R.id.size_spinner) {
+            gameSize = R.string.one.toString()
+        }
+    }
+
+    /**
+     * Start the game based on the chosen mode
+     */
+    private fun proceedGame(){
+        when (this.game) {
+            "local_buzzer" -> { switchToGameWithParameters(BuzzerSetupActivity()) }
+            "local_typing" -> { switchToGameWithParameters(TypingGameActivity()) }
+            "local_lyrics" -> { switchToGameWithParameters(LyricsBelongGameActivity()) }
+        }
+    }
+
+    private fun switchToGameNoParameters(nextActivity: AppCompatActivity){
```

Comment:
consider adding documentation for this function and the following one, mainly what parameters?

Reply:
done!